Update project to .Net Standard 2.0

jgbright opened the pull request:
This change addresses issue #16 .

There are a significant number of changes.  I did not squash the commits so that the individual changes could be reviewed independently.

jgbright commented:
Sure, I'll investigate the appveyor error.

jgbright commented:
I tried both versions before and got the same error with both.  I set it back to the original version to trigger a new build.

ErikSchierboom commented:
Well, I'd be okay with removing the coveralls integration. Could you then also remove the coveralls badge from the README.md file?

jgbright commented:
I fixed running the unit tests and getting the code coverage report.  I believe the reason the coveralls upload is failing is because the repo token environment variable is blank.  I think it is blank because it is marked as secure.

https://www.appveyor.com/docs/build-configuration/#secure-variables

I added logic to conditionally upload to coveralls only if that variable is set.  I think the way it is setup, pull requests cannot use that variable and therefore cannot upload to coveralls.

ErikSchierboom commented on README.md:
```diff
@@ -189,6 +189,13 @@ The library is available on NuGet package available. You can install it using th
   </tr>
+  <tr>
+     <td>2018-04-13</td>
+     <td>?</td>
```
The version should be 2.0.0. To actually have it be that version, you also need to update the assembly info: https://github.com/ErikSchierboom/JsonDotNetCustomContractResolvers/blob/master/src/CustomContractResolvers/Properties/AssemblyInfo.cs

ErikSchierboom commented:
It's already looking very good, just one small comment about the version, and after that it's ready to be merged!

ErikSchierboom commented:
Looking good! I'll merge this tomorrow and then put out a new version.

ErikSchierboom commented:
@jgbright I've just published version 2.0.0 to NuGet: https://www.nuget.org/packages/JsonDotNet.CustomContractResolvers/2.0.0 

Once again, thanks a lot for the help!
